modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb

```diff
@@ -0,0 +1,163 @@
+      end
+
+      def run
+        FileUtils.mkdir_p(temp_directory_path)
+
+        process_submission_files
+
+        temp_directory_path
+      rescue => e
+        handle_error("Failed building submission: #{submission.id}", e, { benefits_intake_uuid: })
+      end
+
+      private
+
+      attr_reader :attachments, :benefits_intake_uuid, :file_path, :include_json_archive, :include_manifest,
+                  :include_text_archive, :metadata, :parent_dir, :submission
+
+      def default_options
+        {
+          attachments: [], # an array of attachment confirmation codes
+          file_path: nil, # file path for the PDF file to be archived
+          include_json_archive: true, # include the form data as a JSON object
+          include_manifest: true, # include a CSV file containing manifest data
+          include_text_archive: true, # include the form data as a text file
+          metadata: {}, # pertinent metadata for original file upload/submission
+          parent_dir: 'vff-simple-forms' # S3 bucket base directory where files live
+        }
+      end
+
+      def process_submission_files
+        write_pdf
+        write_as_json_archive if include_json_archive
+        write_as_text_archive if include_text_archive
+        write_attachments unless attachments.empty?
+        write_manifest if include_manifest
+        write_metadata
+      end
+
+      def write_pdf
+        write_tempfile(submission_pdf_filename, File.read(generate_pdf_content))
+      end
+
+      # TODO: this will be pulled out to be more team agnostic
+      def generate_pdf_content
+        return file_path if file_path
+
+        form_number = SimpleFormsApi::V1::UploadsController::FORM_NUMBER_MAP[submission.form_type]
+        form = "SimpleFormsApi::#{form_number.titleize.gsub(' ', '')}".constantize.new(form_data_hash)
+        filler = SimpleFormsApi::PdfFiller.new(form_number:, form:)
+
+        @file_path = filler.generate(timestamp: submission.created_at)
+        @metadata = SimpleFormsApiSubmission::MetadataValidator.validate(
+          form.metadata,
+          zip_code_is_us_based: form.zip_code_is_us_based
+        )
+
+        form.handle_attachments(file_path) if %w[vba_40_0247 vba_20_10207 vba_40_10007].include? form_number
+
+        @attachments = form.get_attachments if form_number == 'vba_20_10207'
+        @file_path
+      end
+
+      def form_data_hash
+        @form_data_hash ||= JSON.parse(submission.form_data)
+      end
+
+      def submission_pdf_filename
+        @submission_pdf_filename ||= "form_#{form_data_hash['form_number']}.pdf"
+      end
+
+      def error_details(error)
+        "#{error.message}\n\n#{error.backtrace.join("\n")}"
+      end
+
+      def write_as_json_archive
+        write_tempfile('form_json_archive.json', JSON.pretty_generate(form_data_hash))
+      end
+
+      def write_as_text_archive
+        form_data_hash['claim_date'] ||= submission.created_at.iso8601
```

Comment:
Good question, the Benefits team did something similar with the payloads they were creating for remediation so the assumption here is that whoever is doing remediation would be looking for something similar. I honestly don't even know if it's necessary at all given the metadata is also being included. I was planning on (hopefully) getting feedback from the remediation team handling these documents to see what we'd be able to axe. Also, I'm not sold on the naming either but that's snake case of what the Benefits team did.

Reply:
Makes sense. I guess since we don't know maybe we start without and if the remediation team says, "hey we want a `claim_date` field (or similar)" then we can add it back in? But you're right, this info can be found elsewhere, like in the manifest.

Reply:
Yeah, I'm alright taking that bit out. I'll update it!

Reply:
Fixed! 👍